jupyter_client/manager.py

```diff
@@ -616,22 +667,15 @@ async def shutdown_kernel(self, now=False, restart=False):
         self.stop_restarter()
 
         if now:
-            await self._kill_kernel()
+            await self._kill_kernel(restart=restart)
         else:
             self.request_shutdown(restart=restart)
             # Don't send any additional kernel kill messages immediately, to give
             # the kernel a chance to properly execute shutdown actions. Wait for at
             # most 1s, checking every 0.1s.
-            await self.finish_shutdown()
+            await self.finish_shutdown(restart=restart)
 
-        # See comment in KernelManager.shutdown_kernel().
-        overrides_cleanup = type(self).cleanup is not AsyncKernelManager.cleanup
```

Comment:
Seems like this mechanism was removed?

Reply:
Correct.  Since the plan is to release this via a major release, my hope was that previous deprecations would be removed.